application/src/main/java/org/togetherjava/tjbot/features/jshell/JShellCommand.java

```diff
@@ -0,0 +1,298 @@
+
+import org.togetherjava.tjbot.features.CommandVisibility;
+import org.togetherjava.tjbot.features.SlashCommandAdapter;
+import org.togetherjava.tjbot.features.jshell.backend.JShellApi;
+import org.togetherjava.tjbot.features.jshell.render.Colors;
+import org.togetherjava.tjbot.features.utils.RequestFailedException;
+
+import javax.annotation.Nullable;
+
+import java.util.List;
+import java.util.Objects;
+
+public class JShellCommand extends SlashCommandAdapter {
+    private static final String JSHELL_TEXT_INPUT_ID = "jshell";
+    private static final String JSHELL_COMMAND = "jshell";
+    private static final String JSHELL_VERSION_SUBCOMMAND = "version";
+    private static final String JSHELL_EVAL_SUBCOMMAND = "eval";
+    private static final String JSHELL_SNIPPETS_SUBCOMMAND = "snippets";
+    private static final String JSHELL_CLOSE_SUBCOMMAND = "shutdown";
+    private static final String JSHELL_STARTUP_SCRIPT_SUBCOMMAND = "startup-script";
+    private static final String JSHELL_CODE_PARAMETER = "code";
+    private static final String JSHELL_STARTUP_SCRIPT_PARAMETER = "startup-script";
+    private static final String JSHELL_USER_PARAMETER = "user";
+    private static final String JSHELL_INCLUDE_STARTUP_SCRIPT_PARAMETER = "include-startup-script";
+
+    private static final int MIN_MESSAGE_INPUT_LENGTH = 0;
+    private static final int MAX_MESSAGE_INPUT_LENGTH = TextInput.MAX_VALUE_LENGTH;
+
+    private final JShellEval jshellEval;
+
+    /**
+     * Creates an instance of the command.
+     */
+    public JShellCommand(JShellEval jshellEval) {
+        super(JSHELL_COMMAND, "JShell as a command.", CommandVisibility.GUILD);
+
+        this.jshellEval = jshellEval;
+
+        getData().addSubcommands(
+                new SubcommandData(JSHELL_VERSION_SUBCOMMAND, "Get the version of JShell"),
+                new SubcommandData(JSHELL_EVAL_SUBCOMMAND,
+                        "Evaluate java code in JShell, don't fill the optional parameter to access a bigger input box.")
+                            .addOption(OptionType.STRING, JSHELL_CODE_PARAMETER,
+                                    "Code to evaluate. If not supplied, open an inout box.")
+                            .addOption(OptionType.BOOLEAN, JSHELL_STARTUP_SCRIPT_PARAMETER,
+                                    "If the startup script should be loaded, true by default."),
+                new SubcommandData(JSHELL_SNIPPETS_SUBCOMMAND,
+                        "Get the evaluated snippets of the user who sent the command, or the user specified user if any.")
+                            .addOption(OptionType.USER, JSHELL_USER_PARAMETER,
+                                    "User to get the snippets from. If null, get the snippets of the user who sent the command.")
+                            .addOption(OptionType.BOOLEAN, JSHELL_INCLUDE_STARTUP_SCRIPT_PARAMETER,
+                                    "if the startup script should be included, false by default."),
```

Comment:
confusing, there was already a parameter like this?

Reply:
It's different, it's if the user wants to also display the startupt script in the snippets, I don't know if I should just delete this option.

Reply:
u have to rename and reword it, so that its clear and not confusing anymore - then its okay